src/operators.jl

```diff
@@ -322,6 +309,25 @@ function _dot(lhs::AbstractArray{T}, rhs::AbstractArray{S}) where {T,S}
     ret
 end
 
+# this is a generic fallback from stdlib LinearAlgebra that's needed for tests to pass on 0.6
+# for example, dot(::Array{Float64,3}, ::Array{Int,3}) is not defined in 0.6
+if VERSION < v"0.7-"
+    function Base.dot(x::AbstractArray, y::AbstractArray)
```

Comment:
This is type piracy if we define a method for a Base function, it should be with custom types. Can't we implement this in a custom function like `_dot` ?

Reply:
Sorry, I think I just started getting too desparate to get 0.6 working.

The proper solution would probably just be to ensure in the tests that this the arrays are promoted properly first.  I'm not entirely sure why this wasn't causing a problem in the original 0.18.  Let me look into it a bit.  I don't think it would be unreasonable to change the test to ensure that promotion occurs, what do you think?

Reply:
Looking this I am more and more confident that the original 0.18 must have been commiting type piracy.  Regardless, I think the fixes to the tests will be pretty simple so we'll all be much happier with that.

Reply:
Which line of the tests is making it fail ?